```
gcc -Os -fno-strict-aliasing -fno-tree-loop-distribute-patterns -fno-stack-protector -fshort-wchar -Wall -DEFIX64 -DEFI_FUNCTION_WRAPPER -m64 -mno-red-zone  -fpic -I/nix/store/xpbp4fg2mwmpa1q4cai1p76hf0kwrw13-gnu-efi-3.0.14/include/efi -I/nix/store/xpbp4fg2mwmpa1q4cai1p76hf0kwrw13-gnu-efi-3.0.14/include/efi/x86_64 -I/nix/store/xpbp4fg2mwmpa1q4cai1p76hf0kwrw13-gnu-efi-3.0.14/include/efi/protocol -I../include -I../refind -I../libeg -I../mok -I. -I./../include \
      -D__MAKEWITH_GNUEFI -c gnuefi-helper.c -o gnuefi-helper.o
In file included from gnuefi-helper.c:19:
DevicePathUtilities.h:229:3: error: conflicting types for 'EFI_DEVICE_PATH_UTILITIES_PROTOCOL'; have 'struct <anonymous>'
  229 | } EFI_DEVICE_PATH_UTILITIES_PROTOCOL;
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/xpbp4fg2mwmpa1q4cai1p76hf0kwrw13-gnu-efi-3.0.14/include/efi/efi.h:61,
                 from gnuefi-helper.h:24,
                 from gnuefi-helper.c:18:
/nix/store/xpbp4fg2mwmpa1q4cai1p76hf0kwrw13-gnu-efi-3.0.14/include/efi/efidevp.h:648:3: note: previous declaration of 'EFI_DEVICE_PATH_UTILITIES_PROTOCOL' with type 'EFI_DEVICE_PATH_UTILITIES_PROTOCOL' {aka 'struct _EFI_DEVICE_PATH_UTILITIES_PROTOCOL'}
  648 | } EFI_DEVICE_PATH_UTILITIES_PROTOCOL;
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[1]: *** [../Make.common:148: gnuefi-helper.o] Error 1
make[1]: Leaving directory '/build/refind-0.13.2/EfiLib'
make: *** [Makefile:86: gnuefi] Error 2
builder for '/nix/store/gz3r09gr194qlvv04k840336bwjc1kg8-refind-0.13.2.drv' failed with exit code 2
error: build of '/nix/store/gz3r09gr194qlvv04k840336bwjc1kg8-refind-0.13.2.drv' on 'ssh://nix-builder@helsinki-hydra-builder02.wg' failed: builder for '/nix/store/gz3r09gr194qlvv04k840336bwjc1kg8-refind-0.13.2.drv' failed with exit code 2
error: builder for '/nix/store/gz3r09gr194qlvv04k840336bwjc1kg8-refind-0.13.2.drv' failed with exit code 1;
       last 10 log lines:
       > In file included from /nix/store/xpbp4fg2mwmpa1q4cai1p76hf0kwrw13-gnu-efi-3.0.14/include/efi/efi.h:61,
       >                  from gnuefi-helper.h:24,
       >                  from gnuefi-helper.c:18:
       > /nix/store/xpbp4fg2mwmpa1q4cai1p76hf0kwrw13-gnu-efi-3.0.14/include/efi/efidevp.h:648:3: note: previous declaration of 'EFI_DEVICE_PATH_UTILITIES_PROTOCOL' with type 'EFI_DEVICE_PATH_UTILITIES_PROTOCOL' {aka 'struct _EFI_DEVICE_PATH_UTILITIES_PROTOCOL'}
       >   648 | } EFI_DEVICE_PATH_UTILITIES_PROTOCOL;
       >       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > make[1]: *** [../Make.common:148: gnuefi-helper.o] Error 1
       > make[1]: Leaving directory '/build/refind-0.13.2/EfiLib'
       > make: *** [Makefile:86: gnuefi] Error 2
       > builder for '/nix/store/gz3r09gr194qlvv04k840336bwjc1kg8-refind-0.13.2.drv' failed with exit code 2
```
seems to break the `refind` build

--- a/pkgs/development/libraries/gnu-efi/default.nix
+++ b/pkgs/development/libraries/gnu-efi/default.nix
@@ -1,23 +1,16 @@
-{ lib, stdenv, buildPackages, fetchurl, fetchpatch, pciutils }:
+{ lib, stdenv, buildPackages, fetchurl, pciutils
+, gitUpdater }:
 
 with lib;
 
 stdenv.mkDerivation rec {
   pname = "gnu-efi";
-  version = "3.0.11";
+  version = "3.0.14";
 
   src = fetchurl {
     url = "mirror://sourceforge/gnu-efi/${pname}-${version}.tar.bz2";
-    sha256 = "1ffnc4xbzfggs37ymrgfx76j56kk2644c081ivhr2bjkla9ag3gj";
+    sha256 = "tztkOg1Wl9HzltdDFEjoht2AVmh4lXjj4aKCd8lShDU=";
   };
-
-  patches = [
-    # Fix build on armv6l
-    (fetchpatch {
-      url = "https://sourceforge.net/p/gnu-efi/patches/_discuss/thread/25bb273a18/9c4d/attachment/0001-Fix-ARCH-on-armv6-and-other-32-bit-ARM-platforms.patch";
-      sha256 = "0pj03h20g2bbz6fr753bj1scry6919h57l1h86z3b6q7hqfj0b4r";
-    })
-  ];
 
   buildInputs = [ pciutils ];
 
@@ -28,6 +21,12 @@ stdenv.mkDerivation rec {
     "HOSTCC=${buildPackages.stdenv.cc.targetPrefix}cc"
     "CROSS_COMPILE=${stdenv.cc.targetPrefix}"
   ];
+
+  passthru.updateScript = gitUpdater {
+    inherit pname version;
+    # No nicer place to find latest release.
+    url = "https://git.code.sf.net/p/gnu-efi/code";
+  };
 
   meta = with lib; {
     description = "GNU EFI development toolchain";

